blueocean-rest/pom.xml

```diff
@@ -24,4 +24,68 @@
           <artifactId>capability-annotation</artifactId>
       </dependency>
   </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-enforcer-plugin</artifactId>
+                <dependencies>
+                    <dependency>
+                        <groupId>de.skuzzle.enforcer</groupId>
+                        <artifactId>restrict-imports-enforcer-rule</artifactId>
+                        <version>0.7.0</version>
+                    </dependency>
+                </dependencies>
+                <executions>
+                    <execution>
+                        <id>check-no-jenkins-imports</id>
+                        <phase>process-sources</phase>
+                        <goals>
+                            <goal>enforce</goal>
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <!-- All packages can use Extension points -->
+                                <restrictImports implementation="de.skuzzle.enforcer.restrictimports.RestrictImports">
+                                    <basePackage>**</basePackage>
+                                    <bannedImports>
+                                        <bannedImport>jenkins.**</bannedImport>
+                                        <bannedImport>hudson.**</bannedImport>
+                                    </bannedImports>
+                                    <allowedImports>
+                                        <allowedImport>hudson.Extension</allowedImport>
+                                        <allowedImport>hudson.ExtensionPoint</allowedImport>
+                                        <allowedImport>hudson.ExtensionList</allowedImport>
+                                        <allowedImport>hudson.ExtensionListListener</allowedImport>
+                                        <allowedImport>hudson.security.csrf.CrumbExclusion</allowedImport>
+                                        <allowedImport>hudson.util.AdaptedIterator</allowedImport>
+                                    </allowedImports>
+                                    <excludedClasses>
+                                        <excludedClass>io.jenkins.blueocean.rest.factory.*</excludedClass>
```

Comment:
Thats like wildcard import of jenkins models:)  Not sure with this change there is any point in keeping initial design criteria to keep rest model free of Jenkins core model for future proofing.